### PR TITLE
RI-5199 - add deploy pipeline

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

File: redisinsight/ui/src/components/notifications/components/infinite-messages/InfiniteMessages.tsx [hunk 1]
```diff
@@ -286,4 +286,46 @@ export const INFINITE_MESSAGES = {
...
+              {/*      size="s"*/}
+              {/*      color="secondary"*/}
+              {/*      onClick={() => {}}*/}
+              {/*      data-testid="notification-connect-db"*/}
+              {/*    >*/}
+              {/*      Statistics*/}
+              {/*    </EuiButton>*/}
+              {/*  </EuiFlexItem>*/}
+              {/*</EuiFlexGroup>*/}
+            </EuiFlexItem>
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/components/notifications/components/infinite-messages/InfiniteMessages.tsx:325]: please, remove commented code

File: redisinsight/ui/src/components/notifications/components/rdi-deploy-error-content/RdiDeployErrorContent.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      {/*      size="s"*/}
+      {/*      color="warning"*/}
+      {/*      onClick={() => {}}*/}
+      {/*      className="toast-danger-btn"*/}
+      {/*      data-testid="see-errors-btn"*/}
+      {/*    >*/}
+      {/*      Remove API key*/}
+      {/*    </EuiButton>*/}
+      {/*  </EuiFlexItem>*/}
+      {/*</EuiFlexGroup>*/}
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/components/notifications/components/rdi-deploy-error-content/RdiDeployErrorContent.tsx:29]: here, too

File: redisinsight/ui/src/components/notifications/error-messages.tsx [hunk 3]
```diff
@@ -59,5 +60,22 @@ export default {
...
+  RDI_DEPLOY_PIPELINE: (
+    { id, title, message }: { id: string, title?: string, message: string },
+    onClose?: () => void
+  ): Toast => ({
+    id,
+    'data-test-subj': 'toast-error-encryption',
+    color: 'danger',
+    iconType: 'alert',
+    onClose,
+    toastLifeTimeMs: 1000 * 60 * 60 * 12, // 12hr,
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/components/notifications/error-messages.tsx:73]: could you please move 12 hr to constant

File: redisinsight/ui/src/mocks/data/rdi.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+export const MOCK_RDI_PIPELINE_CONFIG = 'connections:\n'
+  + '# Redis data DB connection \n'
+  + '# This section is for configuring the Redis database to which Redis Data Integration will connect to\n'
+  + '  target:\n'
+  + '    # Target type - Redis is the only supported type\n'
+  + '    type: redis\n'
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/mocks/data/rdi.ts:6]: it's better to use template strings

File: redisinsight/ui/src/pages/rdi/pipeline/components/header/RdiPipelineHeader.spec.tsx [hunk 5]
```diff
@@ -12,21 +14,49 @@ beforeEach(() => {
...
   useHistory: () => ({
     push: jest.fn,
   }),
 }))
 
+jest.mock('formik')
+
+beforeEach(() => {
+  cleanup()
+})
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/pipeline/components/header/RdiPipelineHeader.spec.tsx:47]: do we need this cleanup() one more time?

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 6]
```diff
@@ -70,3 +85,33 @@ export function fetchRdiPipeline(
...
+  rdiInstanceId: string,
+  pipeline: IPipeline,
+  onSuccessAction?: () => void,
+  onFailAction?: () => void,
+) {
+  return async (dispatch: AppDispatch) => {
+    try {
+      dispatch(deployPipeline())
+      const { status } = await apiService.post<any>(
+        `rdi/${rdiInstanceId}/pipeline/deploy`,
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:99]: move url please to api constants

File: redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
+  deployPipelineFailure,
   fetchRdiPipeline,
+  deployPipelineAction,
   rdiPipelineSelector,
 } from 'uiSrc/slices/rdi/pipeline'
 import { apiService } from 'uiSrc/services'
-import { addErrorNotification } from 'uiSrc/slices/app/notifications'
+import { addErrorNotification, addInfiniteNotification } from 'uiSrc/slices/app/notifications'
+import { INFINITE_MESSAGES } from 'uiSrc/components/notifications/components'
+import {AnyAction} from "@reduxjs/toolkit";
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts:19]: please, fix quotes and spaces

File: redisinsight/ui/src/templates/rdi-pipeline-page-template/RdiPipelinePageTemplate.tsx [hunk 8]
```diff
@@ -12,15 +17,33 @@ export interface Props {
...
-      <div className={styles.wrapper}>
-        <Navigation />
-        {children}
-      </div>
-    </>
+    <Formik
+      initialValues={{
+        config: data?.config ?? '',
+        jobs: data?.jobs ?? [],
+      }}
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/templates/rdi-pipeline-page-template/RdiPipelinePageTemplate.tsx:35]: I would prefer to move initialValues to the variable

File: redisinsight/ui/src/pages/rdi/pipeline/components/header/RdiPipelineHeader.spec.tsx (hunk 5 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/pages/rdi/pipeline/components/header/RdiPipelineHeader.spec.tsx:47]: I guess nope. Removed. Thank you

File: redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts (hunk 7 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts:19]: Done, thank you

File: redisinsight/ui/src/constants/api.ts [hunk 9]
```diff
@@ -138,7 +138,8 @@ enum ApiEndpoints {
   ANALYTICS_SEND_PAGE = 'analytics/send-page',
 
   RDI_INSTANCES = 'rdi',
-  RDI_PIPELINE = 'rdi/pipeline'
+  RDI_PIPELINE = 'pipeline',
```
Comment (kyle-marcum) [inline redisinsight/ui/src/constants/api.ts:141]: is `pipeline` no longer under `rdi` with this change?

Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/constants/api.ts:141]: pipeline still under rdi. I added new function getRdiUrl.

Comment (kyle-marcum) [reply redisinsight/ui/src/constants/api.ts:141]: 👍

File: redisinsight/ui/src/templates/rdi-pipeline-page-template/RdiPipelinePageTemplate.tsx [hunk 10]
```diff
@@ -1,26 +1,58 @@
-import React from 'react'
+import React, {useEffect, useState} from 'react'
```
Comment (kyle-marcum) [inline redisinsight/ui/src/templates/rdi-pipeline-page-template/RdiPipelinePageTemplate.tsx:1]: Screenshot 2024-01-23 at 4 42 06 PM

**Pull request merged**